### PR TITLE
Signal potentially misconfigured jobs and delete sig-release-x.y-all dashboards

### PR DESCRIPTION
**Pull request by justaugustus**
- kind: Delete kind job for 1.13, which is out of support
- Remove sig-release-x.y-all boards
  These boards have not really proven to be useful to the Release Team
  over time, so we remove them here. Additionally, jobs that land on those
  boards tend to be misconfigured.
- Drop release branch jobs from removed *-all boards
- Move any job on x.y-all dashboards to sig-release-job-config-errors
  Additionally, we remove sig-release-x.y-all from the job generation
  script. We should probably enable a set of default, non-SIG Release
  dashboards for generation in a follow-up.

cc: @alejandrox1 @alenkacz @kubernetes/sig-release-admins @kubernetes/release-team 
/sig release
/area release-eng

### CODE REVIEW

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.17.yaml [hunk 1]
```diff
@@ -130,7 +130,7 @@ periodics:
       resources: {}
 - annotations:
     fork-per-release-generic-suffix: "true"
-    testgrid-dashboards: sig-node-kubelet, sig-release-1.17-all
+    testgrid-dashboards: sig-node-kubelet, sig-release-job-config-errors
     testgrid-tab-name: node-kubelet-features-beta
```
Comment (alejandrox1) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.17.yaml]: Can we save this one?
This one is in a good state (same for previous releases - 1.16, 1.15, etc) https://testgrid.k8s.io/sig-release-1.17-all#node-kubelet-features-beta
And it runs some useful tests `--focus="\[NodeFeature:.+\]" --skip="\[Flaky\]|\[Serial\]"`

Comment (alejandrox1) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.17.yaml]: by "save" i mean move it over to blocking/informing

Comment (justaugustus) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.17.yaml]: @alejandrox1 -- nice catch! Fixed in 1d4af084eb40674bee45e6ddef0655f9ad3812eb.

**Pull request merged**